version/version_test.go

```diff
@@ -1,289 +1,37 @@
-			exp: "18",
-		},
-		// prerelease version does not apply to Dev or Nightly builds
-		{
-			bt:  Dev,
-			exp: "",
-		},
-		{
-			bt:  Nightly,
-			exp: "",
-		},
-	}
-	for i, test := range tests {
-		t.Run(fmt.Sprintf("%d", i), func(t *testing.T) {
-			ver, err := GetPrereleaseVersion(test.tag, test.bt)
-			if test.err && err == nil {
-				t.Fatal("expected error")
-			}
-			if !test.err && err != nil {
-				t.Fatal(err)
-			}
-			if got, want := ver, test.exp; got != want {
-				t.Errorf("bad prelease version: got %q, want %q", got, want)
-			}
-		})
-	}
-}
-
-func TestGetVersion(t *testing.T) {
-	tests := []tagTest{
-		{
-			tag: "",
-			bt:  Dev,
-			err: true, // could not determine base version from tag
-		},
-		{
-			tag: "v2.0.0-rc-2",
-			bt:  RC,
-			err: true, // could not parse prerelease version
-		},
-		{
-			tag: "2.0.0-alpha.17-1",
-			bt:  Alpha,
-			exp: "2.0.0-alpha.17",
-		},
-		{
-			tag: "2.0.0-beta.18-2",
-			bt:  Beta,
-			exp: "2.0.0-beta.18",
-		},
-		{
-			tag: "2.0.0-beta.20-1",
-			bt:  Dev,
-			exp: "2.0.0-dev",
-		},
-		{
-			tag: "2.0.0-beta.20-1",
-			bt:  Nightly,
-			exp: "2.0.0-nightly",
-		},
-	}
-	for i, test := range tests {
-		t.Run(fmt.Sprintf("%d", i), func(t *testing.T) {
-			ver, err := GetVersion(test.tag, test.bt)
-			if test.err && err == nil {
-				t.Fatal("expected error")
-			}
-			if !test.err && err != nil {
-				t.Fatal(err)
-			}
-			if got, want := ver, test.exp; got != want {
-				t.Errorf("bad version: got %q, want %q", got, want)
-			}
-		})
-	}
-}
-
-func TestFullVersion(t *testing.T) {
-	tests := []tagTest{
-		{
-			tag: "v2.0.0-rc-2",
-			bt:  RC,
-			err: true, // could not parse base version
-		},
 		{
-			tag: "2.0.0-alpha.17-1",
-			bt:  Alpha,
-			exp: "2.0.0-alpha.17-1",
+			component: "testing",
+			version:   "1.2.3",
 		},
 		{
-			tag: "2.0.0-beta.18-2",
-			bt:  Beta,
-			exp: "2.0.0-beta.18-2",
+			component: "testing",
+			version:   "1.2.3",
+			buildSHA:  "387c20615518f1325528705e0ef09e4d30d80378",
 		},
 		{
-			tag: "2.0.0-rc.17-1",
-			bt:  Nightly,
-			// nightly build with env var tag
-			env: map[string]string{
-				"SENSU_BUILD_ITERATION": "7",
-			},
-			exp: "2.0.0-nightly-7",
+			component: "testing",
+			version:   "1.2.3",
+			buildDate: "2019-10-09",
+			buildSHA:  "387c20615518f1325528705e0ef09e4d30d80378",
 		},
 	}
-	for i, test := range tests {
-		t.Run(fmt.Sprintf("%d", i), func(t *testing.T) {
-			for k, v := range test.env {
-				if err := os.Setenv(k, v); err != nil {
-					t.Fatal(err)
-				}
-			}
-			defer os.Clearenv()
-			ver, err := FullVersion(test.tag, test.bt)
-			if test.err && err == nil {
-				t.Fatal("expected error")
-			}
-			if !test.err && err != nil {
-				t.Fatal(err)
-			}
-			if got, want := ver, test.exp; got != want {
-				t.Errorf("bad fullversion: got %q, want %q", got, want)
-			}
-		})
+	for _, tt := range tests {
+		Version = tt.version
+		BuildDate = tt.buildDate
+		BuildSHA = tt.buildSHA
+		Println(tt.component)
```

Comment:
Should this be removed?

Reply:
🙈

Reply:
Like discussed, it’s actually required; it uses this pattern to verify the output: https://golang.org/pkg/testing/#hdr-Examples